### DISCUSSION TITLE
[QUESTION] Does Longhorn support xattrs?

### DISCUSSION THREAD

**Discussion by rolnick** [longhorn/longhorn#6782] [General]
## Question
I am in process of hardening kubernetes(RKE2) cluster with SELinux enabled. In most cases everything works fine.
Does Longhorn support xattrs?

## Environment

 - Longhorn version: 1.2.4
 - Kubernetes version: rke2 version v1.23.6+rke2r1
 - Node config
   - OS type and version:  CentOS Linux release 7.9.2009 (Core)
   - CPU per node: 6
   - Memory per node: 24GiB
   - Disk type: SSD
   - Network bandwidth and latency between the nodes: 10Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMWare

## Additional context
Unfortunately when I was trying to start Statefulset with volume: ```accessModes: - ReadWriteMany```, I got an error:
```Error: failed to generate container "ad4c1d879d3bafd3c2a408ff840173a03b0f9b8756f8077ebda698a6b4fd5c10" spec: failed to generate spec: relabel "/var/lib/kubelet/pods/273ccdbf-bd23-4ba6-8cd8-ad6443e35b09/volumes/kubernetes.io~csi/pvc-07143c13-b920-4412-b906-cc42cc4329ca/mount" with "system_u:object_r:svirt_sandbox_file_t:s0:c148,c553" failed: lsetxattr /var/lib/kubelet/pods/273ccdbf-bd23-4ba6-8cd8-ad6443e35b09/volumes/kubernetes.io~csi/pvc-07143c13-b920-4412-b906-cc42cc4329ca/mount: operation not supported```

What is more funny, when I switched to ReadWriteOnce everything worked fine.

**Comment by derekbit** [comment]
RWX volume relies on NFSv4.1. Not sure if this issue is related to the NFS protocol. Need further investigation.

**Comment by rolnick** [comment]
I think I found the issue. I have two disk on my machines, and I created symbolic link /var/lib/kubelet(sda) -> /data//kubelet(sdb).
I removed it, and everything works fine now.
More: https://elixir.bootlin.com/linux/latest/source/fs/xattr.c#L125
